src/types/instantsearch.ts

```diff
@@ -91,11 +92,26 @@ export type NumericRefinement = {
 
 export type Refinement = FacetRefinement | NumericRefinement;
 
+export type Router = {
+  onUpdate(callback: (route: Partial<SearchParameters>) => void): void;
+  read(): UiState;
+  write(route: Partial<SearchParameters>): void;
+  createURL(state: UiState): string;
+  dispose(): Widget['dispose'];
+};
+
+export type StateMapping = {
+  stateToRoute(state: UiState): UiState;
```

Comment:
This is not 100% correct, we should define a `RouteState` generics with a default of `UiState`. 

This is also true for the `routeToState` function.